### PR TITLE
udev link advertizing followup

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/test/test-util.c [hunk 1]
```diff
@@ -13,11 +13,14 @@
 #include "raw-clone.h"
 #include "rm-rf.h"
 #include "string-util.h"
+#include "tests.h"
 #include "util.h"
 
 static void test_align_power2(void) {
         unsigned long i, p2;
 
+        log_info("/* %s */", __func__);
```
Comment (poettering) [inline src/test/test-util.c:22]: maybe turn this into a macro? log_test_func()?

File: src/basic/util.h [hunk 2]
```diff
@@ -148,6 +148,13 @@ static inline int memcmp_safe(const void *s1, const void *s2, size_t n) {
         return memcmp(s1, s2, n);
 }
 
+/* Only for small arrays! Does double evaluation of the argument. */
+#define array_is_zero(arr)                                      \
+        ({                                                      \
+                __typeof__(arr) _zzz = {};                      \
+                memcmp((arr), _zzz, sizeof(_zzz)) == 0;         \
+        })
```
Comment (poettering) [inline src/basic/util.h]: hmm, wouldn't it be better to add memory_all_zeroes() or so as regular function that just iterates through the thing bytewise and checks for non-zero? Sure, would not be as fast, but at least work with all kinds of memory?

Comment (poettering) [reply src/basic/util.h]: maybe copy something like this into our sources somewhere: 

https://github.com/rustyrussell/ccan/blob/master/ccan/mem/mem.c#L92

File: src/test/test-util.c (hunk 1 above)
Comment (keszybz) [reply src/test/test-util.c:22]: It wouldn't save much. In various places there's additional information being printed (for example args for the test function when its called multiple times).

**Comment by keszybz** [comment]
Update to use `memeqzero` (actually `eqzero`). This is similar to how we have `memzero` and `zero`.

**Pull request merged**